app/models/orchestration.rb

```diff
@@ -26,9 +27,14 @@ def run
   # rubocop:enable Rails/SkipsModelValidations
 
   def self.run(kind: :bootstrap)
+    raise OrchestrationOngoing unless runnable?
     Orchestration.create!(kind: kind).tap(&:run)
   end
 
+  def self.runnable?
+    Orchestration.all.none?(&:in_progress?)
```

Comment:
It's always better to use the scopes:

```
irb(main):005:0> Orchestration.all.none?(&:in_progress?)
  Orchestration Load (0.4ms)  SELECT `orchestrations`.* FROM `orchestrations`
=> true
```

as opposed to

```
irb(main):004:0> Orchestration.in_progress.empty?
   (3.5ms)  SELECT COUNT(*) FROM `orchestrations` WHERE `orchestrations`.`status` = 0
=> true
```

With `Orchestration.in_progress.empty?` we don't need to load the records in memory just to check if there's any, and the SQL generated sentence is smarter.

Reply:
updated, thx